.github/workflows/python-client.yml

```diff
@@ -52,10 +55,10 @@ jobs:
           cache-prefix: python-client-tox
       - uses: ./.github/workflows/python-tests
         with:
-          python-version: ${{ matrix.python }}
+          python-version: ${{ matrix.python-version }}
```

Comment:
Having different variable names for the same thing `with.python-version` vs. `matrix.python` is usually a good source of bugs.